connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -1104,7 +1104,9 @@ public void stopConnector(final String connName, final Callback<Void> callback)
                     writeTaskConfigs(connName, Collections.emptyList());
                     configBackingStore.putTargetState(connName, TargetState.STOPPED);
                     // Force a read of the new target state for the connector
-                    refreshConfigSnapshot(workerSyncTimeoutMs);
+                    if (!refreshConfigSnapshot(workerSyncTimeoutMs)) {
+                        throw new ConnectException("Failed to read to end of config topic");
```

Comment:
Agreed, better to log a warning 👍

I was also toying with the idea that we could shift the responsibility of refreshing the view of the config topic from operations that write to the topic, to operations that read from the topic. For example, if we need to check that a connector exists before pausing/resuming/restarting it, then that operation fails if we can't update our view of the topic.

I don't think this is necessary to do in this PR and can be a follow-up. It's also questionable if we should even bother since there's still race conditions that can come from target state updates that originate from non-leader workers.

But it's at least worth some food for thought.

Reply:
Yeah, that does seem to make sense. So, for the stopped state for instance, we'd refresh the config topic view before writing task configs (to ensure that the connector isn't in a stopped state) or resetting connector offsets instead of doing so after writing the stopped target state itself?

Reply:
Yep, exactly!